src/main/java/frc/robot/subsystems/intakeLimelight/IntakeLimelight.java

```diff
@@ -0,0 +1,579 @@
+   * https://www.chiefdelphi.com/t/what-does-limelight-skew-actually-measure/381167/7
+   *
+   * @return Translation2d from the camera to the target. This is camera relative, x is out of the
+   *     camera and positive y is to the left from the camera's point of view.
+   */
+  public Translation2d getTargetTranslation() {
+    return getTargetTranslation(kTargetHeight);
+  }
+
+  /** Gets the object lowest in the image */
+  public Optional<Double> getXOfSmallestY(double[] corners) {
+    // Format: x1 y1 x2 y2 x3 y3 . . .
+    double minY = 10000000;
+    int minYIdx = 0;
+
+    if (corners.length == 0) {
+      return Optional.empty();
+    }
+
+    for (int i = 1; i < corners.length; i += 2) {
+      if (corners[i] < minY) {
+        minY = corners[i];
+        minYIdx = i;
+      }
+    }
+
+    return Optional.of(corners[minYIdx - 1]);
+  }
+
+  public Optional<Double> getObjectHeightPx(double[] corners) {
+    // Format: x1 y1 x2 y2 x3 y3 . . .
+    if (corners.length == 0) {
+      return Optional.empty();
+    }
+
+    double minY = Double.MAX_VALUE;
+    double maxY = Double.MIN_VALUE;
+
+    for (int i = 1; i < corners.length; i += 2) {
+      minY = Math.min(minY, corners[i]);
+      maxY = Math.max(maxY, corners[i]);
+    }
+
+    return Optional.of(maxY - minY);
+  }
+
+  public class NoteDetection {
+    public double latencySec;
+
+    /** Distance from camera */
+    public double distanceMeters;
+
+    /** CCW Yaw Angle */
+    public double yawAngleDeg;
+
+    public NoteDetection(double latencySec, double distanceMeters, double yawAngleDeg) {
+      this.latencySec = latencySec;
+      this.distanceMeters = distanceMeters;
+      this.yawAngleDeg = yawAngleDeg;
+    }
+  }
+
+  /**
+   * Looks for a note
+   *
+   * @return If empty, no note detected. First value is note yaw in degrees (ccw), second value is
+   *     distance in meters.
+   */
+  public Optional<NoteDetection> getNotePos() {
+    // TODO filter low confidence detection in LL dashboard, hopefully
+    if (getPipeline() != pipeline.NOTE_PIPELINE) {
+      setPipeline(pipeline.NOTE_PIPELINE);
+    }
+
+    LimelightHelpers.LimelightResults llresults =
+        LimelightHelpers.getLatestResults(IntakeLimelightConstants.INTAKE_LIMELIGHT_NAME);
+    LimelightTarget_Detector[] targets_Detector = llresults.targetingResults.targets_Detector;
+
+    if (targets_Detector.length == 0) {
+      System.out.println("Didn't see note");
+      return Optional.empty();
+    }
+
+    LimelightTarget_Detector lowestDetection = targets_Detector[0];
+    for (LimelightTarget_Detector detection : targets_Detector) {
+      if (detection.className == "note") {
+        if (detection.ty_pixels > lowestDetection.ty_pixels) { // lower in image = greater y value
+          lowestDetection = detection;
+        }
+      }
+    }
+
+    double angleLimelightToNote =
+        IntakeLimelightConstants.INTAKE_LIMELIGHT_PITCH_DEGREES + lowestDetection.ty;
```

Comment:
I think `INTAKE_LIMELIGHT_PITCH_DEGREES` is defined as 0 parallel to ground, 90 up. How is ty defined?

Reply:
Add units to var name also

Reply:
"Y-coordinate of the center of the target in degrees. Positive-down, center-zero" so 0 is fine, and sign is also correct. (If we have a note 10deg below horizontal, that is +10 in limelight degrees, and if our limelight is angled 10 degrees up, that is +10 in robot degrees; (+10) + (+10) = +20, which I believe is what we want.

I have changed the name of the var.